pkg/rbac/dbAvailability.go

```diff
@@ -0,0 +1,21 @@
+package rbac
+
+import (
+	"net/http"
+
+	"k8s.io/klog/v2"
+)
+
+func (c *Cache) CheckDBAvailability(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// if postgres db is not setup, return error
+		if !c.dbConnInitialized {
+			http.Error(w, "Unable to establish connection with database.", http.StatusServiceUnavailable)
+			return
+		}
+
+		klog.V(6).Info("Connection with database successful!")
```

Comment:
This log message could be misleading because we aren't testing the connection.
I suggest that we remove it and only log on the error path above.
```suggestion
		klog.V(6).Info("Connection with database successful!")
```